tests/test_manager.py

```diff
@@ -372,3 +372,98 @@ def test_request_body_format(self):
         body = manager.save_or_put({"bing": "bong"})[3]
 
         self.assertTrue(body, "<Invoice><bing>bong</bing></Invoice>")
+
+    def test_idempotency_key(self):
+        """
+        A valid idempotency key should be provided as a header on post and
+        put API calls. An invalid idempotency key should raise an exception.
+        """
+        from io import BytesIO
+
+        from xero.utils import generate_idempotency_key
```

Comment:
Imports should always be at the top of the file, unless there's an import order problem (in which case the import order issue should be documented.